leaderboardsgg/leaderboard-site
# LBGG-564: Account confirmation

## What
Enables the user to confirm their account from the link provided in their signup confirmation email.

### Changes
- Updated 
  - API library
- Created
  - `useConfirmAccount` composable
  - components
    - `BasicAlert`
    - `Loader`
  - `/confirm-account` route
- Fixed
  - version requirements in `package`
  - rendering error messages on signup

## Link to Issue

Closes https://github.com/leaderboardsgg/leaderboard-site/issues/564

## Acceptance

### Steps for testing
- Run the latest version of the backend
- Sign up for a new account with an email
- Click the account confirmation link in the email from the sign up
- [ ] Should confirm your account with a success notification
- Use the confirmation link again
- [ ] Should show an error notification after the redirect

### Images

| State | Image |
| --- | --- |
| Attempting to confirm account | ![image](https://github.com/leaderboardsgg/leaderboard-site/assets/17736056/16b60017-2254-4d3f-a454-ace90523d376) |
| Confirmation successful | ![image](https://github.com/leaderboardsgg/leaderboard-site/assets/17736056/a87f8e38-3e8a-4b22-bcfd-baa4b1250118) |
| Confirmation failed | ![image](https://github.com/leaderboardsgg/leaderboard-site/assets/17736056/b2c8518c-bf1e-42c6-bfe5-12d9ffa5a65d) |
